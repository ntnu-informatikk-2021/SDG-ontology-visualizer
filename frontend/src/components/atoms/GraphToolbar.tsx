import { Checkbox, HStack } from '@chakra-ui/react';
import React from 'react';
import SearchBar from './SearchBar';

type GraphToolBarProps = {
  onSubgoalFilter: () => void;
};

const GraphToolBar: React.FC<GraphToolBarProps> = ({ onSubgoalFilter }: GraphToolBarProps) => (
  <HStack bg="cyan.500" borderRadius="lg" p="2" spacing="5">
    <SearchBar limit={5} />
    <Checkbox color="white" size="lg" onChange={onSubgoalFilter}>
      Vis delm√•l
    </Checkbox>
  </HStack>
);

export default GraphToolBar;
