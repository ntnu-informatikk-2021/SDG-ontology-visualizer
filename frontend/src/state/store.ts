import { createStore } from 'redux';
import apiErrorReducer from './reducers/apiErrorReducer';

const store = createStore(apiErrorReducer);

export default store;
