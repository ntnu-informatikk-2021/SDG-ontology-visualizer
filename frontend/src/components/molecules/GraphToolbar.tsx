import { Button, Checkbox, HStack } from '@chakra-ui/react';
import React from 'react';
import { useSelector } from 'react-redux';
import { RootState } from '../../state/store';
import SearchBar from '../atoms/SearchBar';
import ShowDropdown from '../atoms/ShowDropdown';
import CorrelationDropdown from './CorrelationDropdown';

type GraphToolBarProps = {
  onSubgoalFilter: () => void;
  onUnlockNodes: React.Dispatch<React.SetStateAction<boolean>>;
  onEdgeLabelsVisible: React.Dispatch<React.SetStateAction<boolean>>;
};

const GraphToolBar: React.FC<GraphToolBarProps> = ({
  onSubgoalFilter,
  onUnlockNodes,
  onEdgeLabelsVisible,
}: GraphToolBarProps) => {
  const { isFullscreen } = useSelector((state: RootState) => state.fullscreenStatus);

  return (
    <HStack
      bg="cyan.700"
      borderRadius={isFullscreen ? 'none' : 'lg'}
      p="3"
      spacing={[null, null, 2, 5, 10]}
    >
      <SearchBar limit={5} />
      <ShowDropdown onSubgoalFilter={onSubgoalFilter} onEdgeLabelsVisible={onEdgeLabelsVisible} />
      <HStack spacing="10" d={['none', 'none', 'none', 'none', 'flex']}>
        <Checkbox colorScheme="cyan" color="white" size="md" checked onChange={onSubgoalFilter}>
          Vis delmål
        </Checkbox>
        <Checkbox
          defaultIsChecked
          colorScheme="cyan"
          color="white"
          size="md"
          onChange={() => onEdgeLabelsVisible((current) => !current)}
        >
          Vis kanttekst
        </Checkbox>
      </HStack>
      <CorrelationDropdown isPositive />
      <CorrelationDropdown isPositive={false} />
      <Button
        color="cyan.700"
        size="sm"
        minW="8em"
        onClick={() => onUnlockNodes((current) => !current)}
      >
        Lås opp noder
      </Button>
    </HStack>
  );
};

export default GraphToolBar;
