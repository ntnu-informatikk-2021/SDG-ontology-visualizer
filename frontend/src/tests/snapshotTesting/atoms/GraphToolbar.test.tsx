import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import GraphToolbar from '../../../components/molecules/GraphToolbar';
import store from '../../../state/store';

it('renders when there are no items', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <GraphToolbar onSubgoalFilter={() => {}} />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
