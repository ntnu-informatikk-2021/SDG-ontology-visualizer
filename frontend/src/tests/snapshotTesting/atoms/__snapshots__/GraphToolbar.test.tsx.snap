// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there are no items 1`] = `
<div
  className="chakra-stack css-1pyzomu"
>
  <div
    aria-controls="popover-content-2"
    aria-expanded={false}
    aria-haspopup="dialog"
    className="chakra-input__group css-u2vt5l"
    id="popover-trigger-2"
    onClick={[Function]}
  >
    <div
      className="chakra-input__left-addon css-1pwgdj0"
      size="sm"
    >
      <svg
        className="chakra-icon css-tud5pv"
        focusable={false}
        viewBox="0 0 24 24"
      >
        <path
          d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
          fill="currentColor"
        />
      </svg>
    </div>
    <input
      aria-label="Søk i bærekraftsontologien"
      className="chakra-input css-lslgl1"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Søk..."
      value=""
    />
  </div>
  <div
    className="css-0"
    style={
      Object {
        "--popper-arrow-shadow-color": "gray.200",
        "--popper-arrow-size": undefined,
        "visibility": "hidden",
      }
    }
  >
    <section
      className="chakra-popover__content css-1lhufj9"
      id="popover-content-2"
      onBlur={[Function]}
      onKeyDown={[Function]}
      role="dialog"
      style={
        Object {
          "opacity": 0,
          "transform": "scale(0.95) translateZ(0)",
          "transformOrigin": "var(--popper-transform-origin)",
          "visibility": "hidden",
        }
      }
      tabIndex={-1}
    >
      <div
        className="chakra-stack css-5t4008"
      />
    </section>
  </div>
  <div
    className="css-tu7qm9"
  >
    <button
      aria-controls="menu-list-6"
      aria-expanded={false}
      aria-haspopup="menu"
      className="chakra-button chakra-menu__menu-button css-4jegua"
      id="menu-button-6"
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <span
        className="css-a9hmjk"
      >
        Vis
      </span>
      <span
        className="chakra-button__icon css-1oeb2oe"
      >
        <svg
          aria-hidden={true}
          className="chakra-icon css-onkibi"
          focusable={false}
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            fill="currentColor"
          />
        </svg>
      </span>
    </button>
    <div
      className="css-0"
      style={
        Object {
          "visibility": "hidden",
        }
      }
    >
      <div
        aria-orientation="vertical"
        className="chakra-menu__menu-list css-11v8hjs"
        id="menu-list-6"
        onKeyDown={[Function]}
        role="menu"
        style={
          Object {
            "opacity": 0,
            "transform": "scale(0.8) translateZ(0)",
            "transformOrigin": "var(--popper-transform-origin)",
            "visibility": "hidden",
          }
        }
        tabIndex={-1}
      >
        <button
          className="chakra-menu__menuitem css-wihc8h"
          id="menuitem-3"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseMove={[Function]}
          role="menuitem"
          tabIndex={0}
          type="button"
        >
          <label
            checked={true}
            className="chakra-checkbox css-q49uxz"
          >
            <input
              checked={false}
              className="chakra-checkbox__input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              style={
                Object {
                  "border": "0px",
                  "clip": "rect(0px, 0px, 0px, 0px)",
                  "height": "1px",
                  "margin": "-1px",
                  "overflow": "hidden",
                  "padding": "0px",
                  "position": "absolute",
                  "whiteSpace": "nowrap",
                  "width": "1px",
                }
              }
              type="checkbox"
            />
            <span
              aria-hidden={true}
              className="chakra-checkbox__control css-dnty2r"
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
            />
            <span
              className="chakra-checkbox__label css-1oeb2oe"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            >
              Vis delmål
            </span>
          </label>
        </button>
        <button
          className="chakra-menu__menuitem css-wihc8h"
          id="menuitem-4"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseMove={[Function]}
          role="menuitem"
          tabIndex={0}
          type="button"
        >
          <label
            className="chakra-checkbox css-q49uxz"
          >
            <input
              checked={true}
              className="chakra-checkbox__input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              style={
                Object {
                  "border": "0px",
                  "clip": "rect(0px, 0px, 0px, 0px)",
                  "height": "1px",
                  "margin": "-1px",
                  "overflow": "hidden",
                  "padding": "0px",
                  "position": "absolute",
                  "whiteSpace": "nowrap",
                  "width": "1px",
                }
              }
              type="checkbox"
            />
            <span
              aria-hidden={true}
              className="chakra-checkbox__control css-dnty2r"
              data-checked=""
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
            >
              <div
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "height": "100%",
                    "justifyContent": "center",
                    "transform": "none",
                  }
                }
              >
                <svg
                  className="css-f5ntj6"
                  style={
                    Object {
                      "fill": "none",
                      "opacity": 1,
                      "stroke": "currentColor",
                      "strokeDasharray": 16,
                      "strokeDashoffset": 0,
                      "strokeWidth": 2,
                    }
                  }
                  viewBox="0 0 12 10"
                >
                  <polyline
                    points="1.5 6 4.5 9 10.5 1"
                  />
                </svg>
              </div>
            </span>
            <span
              className="chakra-checkbox__label css-1oeb2oe"
              data-checked=""
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            >
              Vis kanttekst
            </span>
          </label>
        </button>
      </div>
    </div>
  </div>
  <div
    className="chakra-stack css-1xvl5m5"
  >
    <label
      checked={true}
      className="chakra-checkbox css-1gefvdb"
    >
      <input
        checked={false}
        className="chakra-checkbox__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        style={
          Object {
            "border": "0px",
            "clip": "rect(0px, 0px, 0px, 0px)",
            "height": "1px",
            "margin": "-1px",
            "overflow": "hidden",
            "padding": "0px",
            "position": "absolute",
            "whiteSpace": "nowrap",
            "width": "1px",
          }
        }
        type="checkbox"
      />
      <span
        aria-hidden={true}
        className="chakra-checkbox__control css-dnty2r"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
      />
      <span
        className="chakra-checkbox__label css-1oeb2oe"
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      >
        Vis delmål
      </span>
    </label>
    <label
      className="chakra-checkbox css-1gefvdb"
    >
      <input
        checked={true}
        className="chakra-checkbox__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        style={
          Object {
            "border": "0px",
            "clip": "rect(0px, 0px, 0px, 0px)",
            "height": "1px",
            "margin": "-1px",
            "overflow": "hidden",
            "padding": "0px",
            "position": "absolute",
            "whiteSpace": "nowrap",
            "width": "1px",
          }
        }
        type="checkbox"
      />
      <span
        aria-hidden={true}
        className="chakra-checkbox__control css-dnty2r"
        data-checked=""
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "height": "100%",
              "justifyContent": "center",
              "transform": "none",
            }
          }
        >
          <svg
            className="css-f5ntj6"
            style={
              Object {
                "fill": "none",
                "opacity": 1,
                "stroke": "currentColor",
                "strokeDasharray": 16,
                "strokeDashoffset": 0,
                "strokeWidth": 2,
              }
            }
            viewBox="0 0 12 10"
          >
            <polyline
              points="1.5 6 4.5 9 10.5 1"
            />
          </svg>
        </div>
      </span>
      <span
        className="chakra-checkbox__label css-1oeb2oe"
        data-checked=""
        onMouseDown={[Function]}
        onTouchStart={[Function]}
      >
        Vis kanttekst
      </span>
    </label>
  </div>
  <button
    aria-controls="menu-list-11"
    aria-expanded={false}
    aria-haspopup="menu"
    className="chakra-button chakra-menu__menu-button css-1li9p89"
    id="menu-button-11"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    <span
      className="css-a9hmjk"
    >
      Positive 
      virkninger
    </span>
    <span
      className="chakra-button__icon css-1oeb2oe"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-onkibi"
        focusable={false}
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          fill="currentColor"
        />
      </svg>
    </span>
  </button>
  <div
    className="css-0"
    style={
      Object {
        "visibility": "hidden",
      }
    }
  >
    <div
      aria-orientation="vertical"
      className="chakra-menu__menu-list css-11v8hjs"
      id="menu-list-11"
      onKeyDown={[Function]}
      role="menu"
      style={
        Object {
          "opacity": 0,
          "transform": "scale(0.8) translateZ(0)",
          "transformOrigin": "var(--popper-transform-origin)",
          "visibility": "hidden",
        }
      }
      tabIndex={-1}
    >
      <button
        className="chakra-menu__menuitem css-wihc8h"
        id="menuitem-7"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
        role="menuitem"
        tabIndex={0}
        type="button"
      >
        <label
          className="chakra-checkbox css-q49uxz"
          type="checkbox"
        >
          <input
            checked={true}
            className="chakra-checkbox__input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            style={
              Object {
                "border": "0px",
                "clip": "rect(0px, 0px, 0px, 0px)",
                "height": "1px",
                "margin": "-1px",
                "overflow": "hidden",
                "padding": "0px",
                "position": "absolute",
                "whiteSpace": "nowrap",
                "width": "1px",
              }
            }
            type="checkbox"
          />
          <span
            aria-hidden={true}
            className="chakra-checkbox__control css-dnty2r"
            data-checked=""
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "height": "100%",
                  "justifyContent": "center",
                  "transform": "none",
                }
              }
            >
              <svg
                className="css-f5ntj6"
                style={
                  Object {
                    "fill": "none",
                    "opacity": 1,
                    "stroke": "currentColor",
                    "strokeDasharray": 16,
                    "strokeDashoffset": 0,
                    "strokeWidth": 2,
                  }
                }
                viewBox="0 0 12 10"
              >
                <polyline
                  points="1.5 6 4.5 9 10.5 1"
                />
              </svg>
            </div>
          </span>
          <span
            className="chakra-checkbox__label css-1oeb2oe"
            data-checked=""
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          >
            Lav
          </span>
        </label>
      </button>
      <button
        className="chakra-menu__menuitem css-wihc8h"
        id="menuitem-8"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
        role="menuitem"
        tabIndex={0}
        type="button"
      >
        <label
          className="chakra-checkbox css-q49uxz"
          type="checkbox"
        >
          <input
            checked={true}
            className="chakra-checkbox__input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            style={
              Object {
                "border": "0px",
                "clip": "rect(0px, 0px, 0px, 0px)",
                "height": "1px",
                "margin": "-1px",
                "overflow": "hidden",
                "padding": "0px",
                "position": "absolute",
                "whiteSpace": "nowrap",
                "width": "1px",
              }
            }
            type="checkbox"
          />
          <span
            aria-hidden={true}
            className="chakra-checkbox__control css-dnty2r"
            data-checked=""
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "height": "100%",
                  "justifyContent": "center",
                  "transform": "none",
                }
              }
            >
              <svg
                className="css-f5ntj6"
                style={
                  Object {
                    "fill": "none",
                    "opacity": 1,
                    "stroke": "currentColor",
                    "strokeDasharray": 16,
                    "strokeDashoffset": 0,
                    "strokeWidth": 2,
                  }
                }
                viewBox="0 0 12 10"
              >
                <polyline
                  points="1.5 6 4.5 9 10.5 1"
                />
              </svg>
            </div>
          </span>
          <span
            className="chakra-checkbox__label css-1oeb2oe"
            data-checked=""
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          >
            Moderat
          </span>
        </label>
      </button>
      <button
        className="chakra-menu__menuitem css-wihc8h"
        id="menuitem-9"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
        role="menuitem"
        tabIndex={0}
        type="button"
      >
        <label
          className="chakra-checkbox css-q49uxz"
          type="checkbox"
        >
          <input
            checked={true}
            className="chakra-checkbox__input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            style={
              Object {
                "border": "0px",
                "clip": "rect(0px, 0px, 0px, 0px)",
                "height": "1px",
                "margin": "-1px",
                "overflow": "hidden",
                "padding": "0px",
                "position": "absolute",
                "whiteSpace": "nowrap",
                "width": "1px",
              }
            }
            type="checkbox"
          />
          <span
            aria-hidden={true}
            className="chakra-checkbox__control css-dnty2r"
            data-checked=""
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "height": "100%",
                  "justifyContent": "center",
                  "transform": "none",
                }
              }
            >
              <svg
                className="css-f5ntj6"
                style={
                  Object {
                    "fill": "none",
                    "opacity": 1,
                    "stroke": "currentColor",
                    "strokeDasharray": 16,
                    "strokeDashoffset": 0,
                    "strokeWidth": 2,
                  }
                }
                viewBox="0 0 12 10"
              >
                <polyline
                  points="1.5 6 4.5 9 10.5 1"
                />
              </svg>
            </div>
          </span>
          <span
            className="chakra-checkbox__label css-1oeb2oe"
            data-checked=""
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          >
            Høy
          </span>
        </label>
      </button>
    </div>
  </div>
  <button
    aria-controls="menu-list-16"
    aria-expanded={false}
    aria-haspopup="menu"
    className="chakra-button chakra-menu__menu-button css-1li9p89"
    id="menu-button-16"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    <span
      className="css-a9hmjk"
    >
      Negative 
      virkninger
    </span>
    <span
      className="chakra-button__icon css-1oeb2oe"
    >
      <svg
        aria-hidden={true}
        className="chakra-icon css-onkibi"
        focusable={false}
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          fill="currentColor"
        />
      </svg>
    </span>
  </button>
  <div
    className="css-0"
    style={
      Object {
        "visibility": "hidden",
      }
    }
  >
    <div
      aria-orientation="vertical"
      className="chakra-menu__menu-list css-11v8hjs"
      id="menu-list-16"
      onKeyDown={[Function]}
      role="menu"
      style={
        Object {
          "opacity": 0,
          "transform": "scale(0.8) translateZ(0)",
          "transformOrigin": "var(--popper-transform-origin)",
          "visibility": "hidden",
        }
      }
      tabIndex={-1}
    >
      <button
        className="chakra-menu__menuitem css-wihc8h"
        id="menuitem-12"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
        role="menuitem"
        tabIndex={0}
        type="button"
      >
        <label
          className="chakra-checkbox css-q49uxz"
          type="checkbox"
        >
          <input
            checked={true}
            className="chakra-checkbox__input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            style={
              Object {
                "border": "0px",
                "clip": "rect(0px, 0px, 0px, 0px)",
                "height": "1px",
                "margin": "-1px",
                "overflow": "hidden",
                "padding": "0px",
                "position": "absolute",
                "whiteSpace": "nowrap",
                "width": "1px",
              }
            }
            type="checkbox"
          />
          <span
            aria-hidden={true}
            className="chakra-checkbox__control css-dnty2r"
            data-checked=""
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "height": "100%",
                  "justifyContent": "center",
                  "transform": "none",
                }
              }
            >
              <svg
                className="css-f5ntj6"
                style={
                  Object {
                    "fill": "none",
                    "opacity": 1,
                    "stroke": "currentColor",
                    "strokeDasharray": 16,
                    "strokeDashoffset": 0,
                    "strokeWidth": 2,
                  }
                }
                viewBox="0 0 12 10"
              >
                <polyline
                  points="1.5 6 4.5 9 10.5 1"
                />
              </svg>
            </div>
          </span>
          <span
            className="chakra-checkbox__label css-1oeb2oe"
            data-checked=""
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          >
            Lav
          </span>
        </label>
      </button>
      <button
        className="chakra-menu__menuitem css-wihc8h"
        id="menuitem-13"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
        role="menuitem"
        tabIndex={0}
        type="button"
      >
        <label
          className="chakra-checkbox css-q49uxz"
          type="checkbox"
        >
          <input
            checked={true}
            className="chakra-checkbox__input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            style={
              Object {
                "border": "0px",
                "clip": "rect(0px, 0px, 0px, 0px)",
                "height": "1px",
                "margin": "-1px",
                "overflow": "hidden",
                "padding": "0px",
                "position": "absolute",
                "whiteSpace": "nowrap",
                "width": "1px",
              }
            }
            type="checkbox"
          />
          <span
            aria-hidden={true}
            className="chakra-checkbox__control css-dnty2r"
            data-checked=""
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "height": "100%",
                  "justifyContent": "center",
                  "transform": "none",
                }
              }
            >
              <svg
                className="css-f5ntj6"
                style={
                  Object {
                    "fill": "none",
                    "opacity": 1,
                    "stroke": "currentColor",
                    "strokeDasharray": 16,
                    "strokeDashoffset": 0,
                    "strokeWidth": 2,
                  }
                }
                viewBox="0 0 12 10"
              >
                <polyline
                  points="1.5 6 4.5 9 10.5 1"
                />
              </svg>
            </div>
          </span>
          <span
            className="chakra-checkbox__label css-1oeb2oe"
            data-checked=""
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          >
            Moderat
          </span>
        </label>
      </button>
      <button
        className="chakra-menu__menuitem css-wihc8h"
        id="menuitem-14"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
        role="menuitem"
        tabIndex={0}
        type="button"
      >
        <label
          className="chakra-checkbox css-q49uxz"
          type="checkbox"
        >
          <input
            checked={true}
            className="chakra-checkbox__input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            style={
              Object {
                "border": "0px",
                "clip": "rect(0px, 0px, 0px, 0px)",
                "height": "1px",
                "margin": "-1px",
                "overflow": "hidden",
                "padding": "0px",
                "position": "absolute",
                "whiteSpace": "nowrap",
                "width": "1px",
              }
            }
            type="checkbox"
          />
          <span
            aria-hidden={true}
            className="chakra-checkbox__control css-dnty2r"
            data-checked=""
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "height": "100%",
                  "justifyContent": "center",
                  "transform": "none",
                }
              }
            >
              <svg
                className="css-f5ntj6"
                style={
                  Object {
                    "fill": "none",
                    "opacity": 1,
                    "stroke": "currentColor",
                    "strokeDasharray": 16,
                    "strokeDashoffset": 0,
                    "strokeWidth": 2,
                  }
                }
                viewBox="0 0 12 10"
              >
                <polyline
                  points="1.5 6 4.5 9 10.5 1"
                />
              </svg>
            </div>
          </span>
          <span
            className="chakra-checkbox__label css-1oeb2oe"
            data-checked=""
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          >
            Høy
          </span>
        </label>
      </button>
    </div>
  </div>
  <button
    className="chakra-button css-12ne2ai"
    onClick={[Function]}
    type="button"
  >
    Lås opp noder
  </button>
</div>
`;
